tests/MomentoRedisClientTest.php

```diff
@@ -897,6 +897,168 @@ public function testZScoreOnExistingMember(): void
+    }
+
+    public function testZunionstore_WithMaxAggregation(): void
+    {
+        $key1 = uniqid();
+        $key2 = uniqid();
+        $dst = uniqid();
+
+        // Set up sorted sets
+        self::$client->zAdd($key1, 1.0, 'a', 2.0, 'b');
+        self::$client->zAdd($key2, 2.0, 'a', 3.0, 'b', 4.0, 'c');
+
+        // Perform zunionstore with max aggregation
+        $result = self::$client->zunionstore($dst, [$key1, $key2], null, 'max');
+
+        // Validate result count
+        $this->assertEquals(3, $result, "zunionstore should return the correct count of unioned elements");
+
+        // Validate the content of the destination set
+        $storedValues = self::$client->zRevRange($dst, 0, -1, true);
+        $expected = [
+            'c' => 4.0,
+            'b' => 3.0,  // max(2.0 from key1, 3.0 from key2)
+            'a' => 2.0,  // max(1.0 from key1, 2.0 from key2)
+        ];
+        $this->assertEquals($expected, $storedValues, "The stored result in $dst should match the expected max values.");
+    }
+
+    public function testZunionstore_WithMinAggregation(): void
+    {
+        $key1 = uniqid();
+        $key2 = uniqid();
+        $dst = uniqid();
+
+        // Set up sorted sets
+        self::$client->zAdd($key1, 1.0, 'a', 2.0, 'b');
+        self::$client->zAdd($key2, 2.0, 'a', 3.0, 'b', 4.0, 'c');
+
+        // Perform zunionstore with min aggregation
+        $result = self::$client->zunionstore($dst, [$key1, $key2], null, 'min');
+
+        // Validate result count
+        $this->assertEquals(3, $result, "zunionstore should return the correct count of unioned elements");
+
+        // Validate the content of the destination set
+        $storedValues = self::$client->zRevRange($dst, 0, -1, true);
+        $expected = [
+            'c' => 4.0,  // No other value to compare to
+            'b' => 2.0,  // min(2.0 from key1, 3.0 from key2)
+            'a' => 1.0,  // min(1.0 from key1, 2.0 from key2)
+        ];
+        $this->assertEquals($expected, $storedValues, "The stored result in $dst should match the expected min values.");
+    }
+
+    public function testZunionstore_WithInvalidWeights_ShouldThrowException(): void
+    {
+        if (self::$isRedisBackendTest) {
+            $this->markTestSkipped("This test is only for Momento client");
+        }
+        $this->expectException(\InvalidArgumentException::class);
+
+        $key1 = uniqid();
+        $key2 = uniqid();
+        $dst = uniqid();
+
+        // Set up sorted sets
+        self::$client->zAdd($key1, 1.0, 'a');
+        self::$client->zAdd($key2, 2.0, 'b');
+
+        // Attempt zunionstore with invalid weights (mismatched count)
+        self::$client->zunionstore($dst, [$key1, $key2], [1.0]);
+    }
+
+    public function testZunionstore_WithInvalidAggregate_ShouldThrowException(): void
+    {
+        if (self::$isRedisBackendTest) {
+            $this->markTestSkipped("This test is only for Momento client");
+        }
```

Comment:
I timedboxed detecting what phpredis does in this scenario and could not get it to catch the exception.